### ISSUE TITLE
Providing a logical xor operator

### ISSUE THREAD

**Issue by DarkDrek** [pvdrz/pijama_legacy#39]
Strictly speaking logical xor is not necessary, it can be implemented with `&&` and `||`, but it would allow writing more compact expression when it is needed.
I would contrary to most other languages use `^^` for this operator to be symmetric to `&&` and `||`.

**Comment by pvdrz** [comment]
I like the symmetry but iirc `x != y` is equivalent to the xor when x and y are booleans.

**Comment by DarkDrek** [comment]
> I like the symmetry but iirc `x != y` is equivalent to the xor when x and y are booleans.

That is correct :1st_place_medal:. I would still favor adding it as operator all the way down or at least as syntax sugar / transform it to `NotEqual` during parsing. (This may cause problem when we generate error messages)

**Comment by pvdrz** [comment]
> > I like the symmetry but iirc `x != y` is equivalent to the xor when x and y are booleans.
> 
> That is correct 1st_place_medal. I would still favor adding it as operator all the way down or at least as syntax sugar / transform it to `NotEqual` during parsing. (This may cause problem when we generate error messages)

I'm not totally convinced of either solution. Adding it as sugar would require to do some special type checking just to be sure it's applied to booleans, and putting it all the way down will add an small overhead during execution.
